### DISCUSSION TITLE
How to use multiple graphs with built in networkx functions

### DISCUSSION THREAD

**Discussion by gmelasecca** [networkx/networkx#6650] [Q&A]
Hello All, 
    I am new to networkx and am really impressed at the capabilities. I will preface this with I am not an expert in graph theory or graph science, etc. Here is my question/problem that I am trying to help resolve.

We are a multi datacenter, customer/tenant/systems and services for a cloud business. We have a networkx DiGraph() made up of **_nodes:edges:attributes_** built from our application layer (virtual) --> virtual infrastructure --> physical infrastructure/hardware (_I am not including the storage and logical network devices inbetween for simplicity_). currently the graph only consists of around 3 million nodes and 3.5 million edges (_not real numbers_) in a single DiGraph(). what I am looking to understand is, does networkx provide the ability to search directionally between multiple DiGraphs()?

scenario:
if I have 3 DiGraph's
1. G1 = nx.DiGraph(source="application")
2. G2 = nx.DiGraph(source="virtual_infra")
3. G3 = nx.DiGraph(source="physical_infra")

lets say I want to run a simple function like dfs_edges, list(nx.dfs_edges(G1, source="application_node_1"). I know that this will search graph G1 but what if I wanted to search all graphs because I know that G1 contains the application nodes and those application nodes should connect to edges (nodes, correct?) in the G2 graph which is virtual infrastructure, which means that G2 needs to also search G3 for physical infrastrucutre which would return me a directional representation from app --> virtual --> physical nodes and edges.

another example would be if I have the same 3 digraphs and wanted to find the shortest_path from app --> physical infrastructure node, I would need to search the path from G1 --> G2 --> G3 so how can I do this with networkx?

I hope this. makes sense and I appreciate any feedback and or answers to help with this problem.

**Comment by rossbar** [comment]
It sounds like you might want to consider reformulating the problem to better capture the relationships you want to probe. AFAIK there is no way to e.g. perform a search over multiple independent graphs. If you can structure your problem in such a way that information about the "layers" is all contained within a single graph (the flexibility of the node/edge attributes should help), you may be able to apply the built-in functions for your application.
